e2e/utils/test-utils.ts

```diff
@@ -1,9 +1,98 @@
-import {Page} from 'puppeteer';
+import {JSHandle, Page} from 'puppeteer';
 import Checkbox from 'app/element/checkbox';
 import RadioButton from 'app/element/radiobutton';
 import TextOptions from 'app/element/text-options';
 import Textarea from 'app/element/textarea';
 import Textbox from 'app/element/textbox';
+import GoogleLoginPage from 'app/page/google-login';
+import HomePage, {LABEL_ALIAS} from 'app/page/home-page';
+import Link from '../app/element/link';
+import {waitForText} from './wait-utils';
+
+export async function signIn(page: Page) {
+  const loginPage = new GoogleLoginPage(page);
+  await loginPage.login();
+  // this element exists in DOM after user has logged in
+  await page.waitFor(() => document.querySelector('app-signed-in') !== null);
+  const homePage = new HomePage(page);
+  await homePage.waitForLoad();
+}
+
+/**
+ * <pre>
+ * Wait until spinner stops spinning, usually it indicate page is ready.
+ * </pre>
+ */
+export async function waitUntilNoSpinner(page: Page, customTimeout: number = 60000): Promise<void> {
+  // wait maximum 1 second for either spinner to show up
+  const spinSelector = '.spinner, svg';
+  let spinner: JSHandle;
+  try {
+    spinner = await page.waitFor((selector) => {
+      return document.querySelectorAll(selector).length > 0
+    }, {timeout: 1000}, spinSelector);
+  } catch (err) {
+    console.info('waitUntilNoSpinner does not find any spin elements.');
+  }
+  const jValue = await spinner.jsonValue();
+
+  // wait maximum 90 seconds for spinner disappear if spinner existed
+  const spinAnimationSelector = 'svg[style*="spin"], .spinner:empty';
+  // const startTime = performance.now();
+  try {
+    if (jValue) {
+      await page.waitFor((selector) => {
+        return document.querySelectorAll(selector).length === 0;
+      }, {timeout: customTimeout}, spinAnimationSelector);
+    }
+  } catch (err) {
+    throw new Error(err);
+  }
+  // final 1 second wait for page render to finish
+  if (jValue) {
+    await page.waitFor(1000);
+  }
+}
+
+export async function handleRecaptcha(page: Page) {
+  const css = '[id="recaptcha-anchor"][role="checkbox"]';
+  await page.frames().find(async (frame) => {
+    for(const childFrame of frame.childFrames()) {
+      const recaptcha = await childFrame.$$(css);
+      if (recaptcha.length > 0) {
+        await recaptcha[0].click();
```

Comment:
Is this trying to pass the Turing Test by clicking where it says "I am not a robot."? Maybe we should add wobbling to the mouse movement and click off center instead of right in the middle.

I don't suppose there's any way to whitelist our test accounts for captcha bypass.